labs/b5.md

```diff
@@ -0,0 +1,373 @@
+
+
+
+This diagram provides a visual breakdown of CIDR addressing
+![alt text][CIDR]
+
+[CIDR]: https://i.imgur.com/dEgcX0R.png "CIDR addressing example courtesy of CS168"
+
+In order to route IP packets, devices have what is known as a routing table. Routing entries are stored in the routing table and they are essentially rules that tell the device how packets should be forwarded based on IP. A routing entry specifies a subnet and the interface that corresponds to that entry. The device chooses an entry with a subnet that is most specific to a given packet and forwards it out the interface on that entry.
+
+Routing tables are usually to also have a default gateway. This serves as the default catch all for packets in the absence of a more specific matching entry.
+
+Take this routing table for example.
+```
+default via 10.0.2.2 dev eth0
+10.0.2.0/24 dev eth0  proto kernel  scope link  src 10.0.2.15
+10.0.2.128/25 dev eth0  proto kernel  scope link  src 10.0.2.15
+192.168.162.0/24 dev eth1  proto kernel  scope link  src 192.168.162.162
+```
+A packet destined for `8.8.8.8` would be forwarded out eth0, the default gateway.
+A packet destined for `10.0.2.1` would be forwarded according to second entry, out of eth0.
+A packet destined for `10.0.2.254` would be forwarded according to third entry, out of eth0.
+A packet destined for `192.168.162.254` would be forwarded according to the fourth entry, out of eth1.
+
+## Private IP addresses and NAT
+Often you'll see a device with an IP address of `192.168.x.x`, and let's say this device is connected to machine A. Pinging that address from A succeeds.  Your friend on machine B, who communicates with you over the internet but isn't connected to your local network, tries to ping that address as well and fails. Now you might be asking ...
+
+
+![alt text][ytho]
+
+[ytho]: http://i0.kym-cdn.com/entries/icons/original/000/022/978/yNlQWRM.jpg
+
+[private]: https://en.wikipedia.org/wiki/IP_address#Private_addresses
+This is because `192.168.0.0/16` is a private network block. [Certain blocks][private] of IP addresses have been designated by the Internet Engineering Task Force (IETF) to be reserved as private address spaces. Private address are especially useful for devices on a local network that only need to communicate on the local network and do not need a public IP address. This is not to say that if you have a private IP address that you cannot reach the wider net or vice versa.
+
+Network address translation or (NAT) is a procedure by which an IP address is mapped into another network address during routing. For example we could map a private IP address to a public one in order to allow a device on the local private network to communicate with cat picture servers out in the wilds of the web.
+
+Take a look at this diagram
```

Comment:
I'm not sure this text really helps much, it also gets pushed off to the right of the image where I think it would be more useful above/below:

<img width="749" alt="screen shot 2018-02-27 at 6 11 56 pm" src="https://user-images.githubusercontent.com/1523594/36766201-b8bb78b0-1be9-11e8-802b-a9a6fa21871c.png">